<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerDrrImageComputationModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerDrrImageComputationModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>535</width>
    <height>951</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DRR Image Computation</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="2" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="5" column="0" colspan="2">
      <widget class="qSlicerDrrImageComputationPlastimatchParametersWidget" name="PlastimatchParametersWidget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="ctkCheckBox" name="CheckBox_ShowDrrMarkups">
       <property name="text">
        <string>Show DRR markups</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <property name="indicatorIcon">
        <iconset resource="../../../../Slicer/Libs/MRML/Widgets/Resources/qMRMLWidgets.qrc">
         <normaloff>:/Icons/VisibleOn.png</normaloff>
         <normalon>:/Icons/VisibleOff.png</normalon>:/Icons/VisibleOn.png</iconset>
       </property>
       <property name="indicatorIconSize">
        <size>
         <width>21</width>
         <height>21</height>
        </size>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="PushButton_ComputeDrr">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Compute DRR image</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="ctkCollapsibleButton" name="CollapsibleButton_ReferenceInput">
       <property name="text">
        <string>Reference input nodes</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>RT beam:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_CtVolume">
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLScalarVolumeNode</string>
           </stringlist>
          </property>
          <property name="showChildNodeTypes">
           <bool>false</bool>
          </property>
          <property name="noneEnabled">
           <bool>true</bool>
          </property>
          <property name="addEnabled">
           <bool>false</bool>
          </property>
          <property name="removeEnabled">
           <bool>false</bool>
          </property>
          <property name="noneDisplay">
           <string>Please select a CT volume node</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_Camera">
            <property name="nodeTypes">
             <stringlist>
              <string>vtkMRMLCameraNode</string>
             </stringlist>
            </property>
            <property name="showChildNodeTypes">
             <bool>false</bool>
            </property>
            <property name="noneEnabled">
             <bool>true</bool>
            </property>
            <property name="addEnabled">
             <bool>false</bool>
            </property>
            <property name="removeEnabled">
             <bool>false</bool>
            </property>
            <property name="noneDisplay">
             <string>Please select a 3D Camera node</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="PushButton_UpdateBeamFromCamera">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Update beam</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="ctkCollapsibleGroupBox" name="CollapsibleGroupBox_VolumeDirectionToLpsTransformMatrix">
          <property name="title">
           <string>Volume direction to LPS transform matrix</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="collapsed">
           <bool>true</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_10">
             <item row="0" column="0">
              <widget class="ctkMatrixWidget" name="MatrixWidget_VolumeDirectionToLpsTransformMatrix">
               <property name="minimum">
                <double>-1.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1.000000000000000</double>
               </property>
               <property name="decimals">
                <number>4</number>
               </property>
               <property name="singleStep">
                <double>0.000100000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="PushButton_ApplyToVolumeTransform">
             <property name="text">
              <string>Apply transform to volume</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_RtBeam">
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLRTBeamNode</string>
           </stringlist>
          </property>
          <property name="showChildNodeTypes">
           <bool>false</bool>
          </property>
          <property name="noneEnabled">
           <bool>true</bool>
          </property>
          <property name="addEnabled">
           <bool>false</bool>
          </property>
          <property name="removeEnabled">
           <bool>false</bool>
          </property>
          <property name="noneDisplay">
           <string>Please select a RTBeam node</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>CT volume:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Camera:</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="ctkCollapsibleButton" name="CollapsibleButton_GeometryBasicParameters">
       <property name="text">
        <string>Geometry basic parameters</string>
       </property>
       <property name="collapsed">
        <bool>false</bool>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Isocenter to imager distance:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_8">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Imager resolution (columns, rows):</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QGroupBox" name="GroupBox_ImageWindowParameters">
            <property name="title">
             <string>Image window parameters</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Columns</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="ctkRangeWidget" name="RangeWidget_ImageWindowColumns">
               <property name="decimals">
                <number>0</number>
               </property>
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1999.000000000000000</double>
               </property>
               <property name="minimumValue">
                <double>0.000000000000000</double>
               </property>
               <property name="maximumValue">
                <double>1999.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Rows</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="ctkRangeWidget" name="RangeWidget_ImageWindowRows">
               <property name="decimals">
                <number>0</number>
               </property>
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1999.000000000000000</double>
               </property>
               <property name="minimumValue">
                <double>0.000000000000000</double>
               </property>
               <property name="maximumValue">
                <double>1999.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_9">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Imager spacing in mm (columns, rows):</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="ctkCoordinatesWidget" name="CoordinatesWidget_ImagerResolution">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="dimension">
             <number>2</number>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>128.000000000000000</double>
            </property>
            <property name="maximum">
             <double>2000.000000000000000</double>
            </property>
            <property name="coordinates" stdset="0">
             <string>2000,2000</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="ctkSliderWidget" name="SliderWidget_IsocenterImagerDistance">
            <property name="maximum">
             <double>999.899999999999977</double>
            </property>
            <property name="value">
             <double>300.000000000000000</double>
            </property>
            <property name="suffix">
             <string> mm</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="ctkCoordinatesWidget" name="CoordinatesWidget_ImagerSpacing">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="dimension">
             <number>2</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="coordinates" stdset="0">
             <string>0.25,0.25</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="ctkCollapsibleButton" name="CollapsibleButton_ProjectionOnImagerPlane">
       <property name="text">
        <string>Markups perspective projection on the imager plane</string>
       </property>
       <property name="collapsed">
        <bool>true</bool>
       </property>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="3" column="0">
           <widget class="QTableWidget" name="TableWidget_ProjectedPointsCoordinates">
            <column>
             <property name="text">
              <string>Original label</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>R</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>A</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>S</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Width</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Height</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Column</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Row</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Status</string>
             </property>
            </column>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="qSlicerSimpleMarkupsWidget" name="SimpleMarkupsWidget_PointCoordinates">
            <property name="enterPlaceModeOnNodeChange" stdset="0">
             <bool>false</bool>
            </property>
            <property name="optionsVisible" stdset="0">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QGroupBox" name="GroupBox_ProcessControlPoints">
            <property name="title">
             <string>Process control points</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_9">
             <item row="0" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QRadioButton" name="RadioButton_ProjectAllControlPoints">
                 <property name="text">
                  <string>All</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">ButtonGroup_ProjectControlPoints</string>
                 </attribute>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="RadioButton_ProjectSelectedControlPoints">
                 <property name="text">
                  <string>Selected</string>
                 </property>
                 <attribute name="buttonGroup">
                  <string notr="true">ButtonGroup_ProjectControlPoints</string>
                 </attribute>
                </widget>
               </item>
              </layout>
             </item>
             <item row="1" column="0">
              <widget class="QCheckBox" name="CheckBox_CreateValidProjectionsMarkupsNode">
               <property name="text">
                <string>Create a new markups node for valid projections</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QPushButton" name="PushButton_ProjectControlPoints">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Project control point</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="PushButton_ClearProjectedTableWidget">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Clear projection table</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="CheckBox_CreateValidProjectionsTableNode">
               <property name="text">
                <string>Create a new table node for valid projections</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="qMRMLNodeComboBox" name="MRMLNodeComboBox_ParameterSet">
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLDrrImageComputationNode</string>
        </stringlist>
       </property>
       <property name="noneDisplay">
        <string>Please select a node</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Parameter set:</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCheckBox</class>
   <extends>QCheckBox</extends>
   <header>ctkCheckBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCoordinatesWidget</class>
   <extends>QWidget</extends>
   <header>ctkCoordinatesWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkMatrixWidget</class>
   <extends>QWidget</extends>
   <header>ctkMatrixWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkRangeWidget</class>
   <extends>QWidget</extends>
   <header>ctkRangeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerSimpleMarkupsWidget</class>
   <extends>qSlicerWidget</extends>
   <header>qSlicerSimpleMarkupsWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerDrrImageComputationPlastimatchParametersWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerDrrImageComputationPlastimatchParametersWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../Slicer/Libs/MRML/Widgets/Resources/qMRMLWidgets.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>qSlicerDrrImageComputationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_ParameterSet</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerDrrImageComputationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_RtBeam</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerDrrImageComputationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_CtVolume</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerDrrImageComputationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MRMLNodeComboBox_Camera</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>285</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerDrrImageComputationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>SimpleMarkupsWidget_PointCoordinates</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>469</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="ButtonGroup_ProjectControlPoints"/>
 </buttongroups>
</ui>
